src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs

```diff
@@ -452,53 +434,59 @@ private void CreateFromBinaryForm(byte[] binaryForm, int offset)
 
             if (binaryForm[offset + 1] > MaxSubAuthorities)
             {
-                throw new ArgumentException(
-                    SR.Format(SR.IdentityReference_InvalidNumberOfSubauthorities, MaxSubAuthorities),
-nameof(binaryForm));
+                throw new ArgumentException(SR.Format(SR.IdentityReference_InvalidNumberOfSubauthorities, MaxSubAuthorities), nameof(binaryForm));
             }
 
+
             //
             // Make sure the buffer is big enough
             //
 
-            int Length = 1 + 1 + 6 + 4 * binaryForm[offset + 1];
+            int subAuthoritiesLength = binaryForm[offset + 1];
 
-            if (binaryForm.Length - offset < Length)
+            if (subAuthoritiesLength > MaxSubAuthorities)
             {
-                throw new ArgumentException(
-                    SR.ArgumentOutOfRange_ArrayTooSmall,
-nameof(binaryForm));
+                throw new ArgumentOutOfRangeException(
+                    "subAuthoritiesLength",
```

Comment:
Looking at this now, isn't it redundant to the previous throw?  Looks like the `subAuthoritiesLength` local should just be moved up one block.

I was about to say that this should just be an ArgumentException for nameof(binaryForm) when I happened to notice that was exactly the previous block.

Reply:
Yup, duplication, reduced.